sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -0,0 +1,347 @@
+}
+```
+
+## Authenticating a service principal with a client secret
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `ClientSecretCredential`. There's also [a runnable sample](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/samples/typescript/src/clientSecretCredential.ts) to create a Key Vault key client you can copy-paste.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with client secret.
+ */
+function withClientSecretCredential() {
+  const credential = new ClientSecretCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<YOUR_CLIENT_SECRET>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a service principal with environment credentials
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `EnvironmentCredential`. The `EnvironmentCredential` looks for well-known environment variable names to determine how it should authenticate. It effectively acts as a wrapper for the `ClientSecretCredential`, `ClientCertificateCredential` or `UsernamePasswordCredential` depending on which environment variables are present.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withEnvironmentCredential() {
+  let credential = new EnvironmentCredential();
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a service principal with a client certificate
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `ClientCertificateCredential`.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withClientCertificateCredential() {
+  let credential = new ClientCertificateCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<PATH_TO_CERTIFICATE>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account with device code flow
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `DeviceCodeCredential`.
```

Comment:
I'd really like to call out as to how this credential is the one with least amount of set up involved.